test/hub/mocks/MockVaults.sol

```diff
@@ -130,19 +123,22 @@ contract MockVaults is Test, Auth, IAdapter {
     }
 
     function resetMessages() external {
-        delete lastChainDestinations;
         delete lastMessages;
     }
 
     function messageCount() external view returns (uint256) {
         return lastMessages.length;
     }
 
-    function popMessage() external returns (bytes memory) {
+    function popMessage() external returns (bytes memory message) {
         require(lastMessages.length > 0, "mockVaults/no-msgs");
 
-        bytes memory popped = lastMessages[lastMessages.length - 1];
+        message = lastMessages[0];
+
+        for (uint256 i = 1; i < lastMessages.length; i++) {
+            lastMessages[i - 1] = lastMessages[i];
+        }
+
         lastMessages.pop();
-        return popped;
     }
```

Comment:
FYI. I've modified the `popMessage()` method to perform instead a pop first, in order to test the messages in order as they arrive